core/contracts/financial_templates/PricelessPositionManager.sol

```diff
@@ -51,26 +59,41 @@ contract PricelessPositionManager is Testable {
      */
     uint public withdrawalLiveness;
 
-    constructor(uint _expirationTimestamp, uint _withdrawalLiveness, address collateralAddress, bool _isTest)
-        public
-        Testable(_isTest)
-    {
+    constructor(
+        uint _expirationTimestamp,
+        uint _withdrawalLiveness,
+        address _collateralAddress,
+        bool _isTest,
+        address _finderAddress,
+        bytes32 _priceFeedIdentifier
+    ) public Testable(_isTest) {
         expirationTimestamp = _expirationTimestamp;
         withdrawalLiveness = _withdrawalLiveness;
-        collateralCurrency = IERC20(collateralAddress);
+        collateralCurrency = IERC20(_collateralAddress);
+        // TODO: add parameters to register the synthetic token's name and symbol.
+        // TODO: add the collateral requirement. This is needed at settlement and at dispute resolution.
+        // TODO: validate the input of the finder/price identifier inputs.
         Token mintableToken = new Token();
         tokenCurrency = ExpandedIERC20(address(mintableToken));
+        finder = Finder(_finderAddress);
+        priceIdentifer = _priceFeedIdentifier;
     }
 
     modifier onlyPreExpiration() {
         require(getCurrentTime() < expirationTimestamp, "Cannot operate on a position past its expiry time");
         _;
     }
 
+    modifier onlyPostExpiration() {
+        require(getCurrentTime() >= expirationTimestamp, "Cannot operate on a position before its expiry time");
+        _;
+    }
+
     /**
      * @notice Transfers ownership of the caller's current position to `newSponsorAddress`. The address
      * `newSponsorAddress` isn't allowed to have a position of their own before the transfer.
      */
+    // TODO: transfer should not work if there is a pending withdraw
```

Comment:
I know you've highlighted as a TODO, but why not implement it now? It's a one liner right?

`require(positionData.requestPassTimestamp == 0, "Cannot transfer with a pending withdrawal request");`

Reply:
It's for the next PR. I didnt want to add functionality over and above DVM integration here :)